### PR TITLE
Provide full API URL of Saleor to app iframe, add test

### PR DESCRIPTION
**Pull request by lkostrowski**
I want to merge this change because...

https://github.com/saleor/saleor-dashboard/issues/2387

tldr - apps should receive full URL of API, so they don't need to guess where the API is

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

**Review by przlada — APPROVED** [review]

File: src/apps/urls.ts [hunk 1]
```diff
@@ -117,15 +118,28 @@ export const customAppAddUrl = customAppAddPath;
...
+    {
+      /**
+       * @deprecated - domain will be removed in favor of saleorApiUrl.
+       * Current hostname (used as domain) can be extracted from full URL
+       *
+       * Difference will be:
+       * shop.saleor.cloud -> https://shop.saleor.cloud/graphql/
+       */
+      domain: shopDomainHost,
+      saleorApiUrl: getApiUrl(),
```
Comment (mmiszy) [inline src/apps/urls.ts:137]: `shopDomainHost` and `getApiUrl()` should match, correct? We can even use `getApiUrl()` to generate `shopDomainHost`.
I don't understand why there's such a different approach to obtaining these two values – one is passed as parameter and the other is imported from `@saleor/config`.

Comment (lkostrowski) [reply src/apps/urls.ts:137]: `getApiUrl` is static on the build time so I don't see a reason it should be passed as a parameter

`shopDomainHost` will not match `getApiUrl` because one is a hostname, other is a full domain. Hence we pass both of them to iframe. Yes we can resolve hostman from saleorApiUrl but we want to keep compatibility for now

**Review by orzechdev — APPROVED** [review]
